src/modules/Builder/components/Builder/Builder.jsx

```diff
@@ -39,20 +40,28 @@ export default class Builder extends Component {
     }
 
     onNextBuildState() {
+        this.builderCallback = this.builderCallback.bind(this);
         if (!this.waitExecutingState) {
             this.waitExecutingState = true;
-
             this.currentStateRef.current.executeState((stateData) => {
-                this.processStateData(stateData);
+                this.builderCallback(stateData);
             });
         }
     }
 
+    builderCallback(stateData) {
+        this.processStateData(stateData);
+    }
+
     processStateData(data) {
         const { currentRoute, prevRouteIndex } = this.state;
         switch (data ? data.state : "") {
             case "initialization":
                 this.buildUUID = data.buildUUID;
+                this.setState(({ device: data.device, os: data.os }));
```

Comment:
Let's set the state asynchronously everywhere. Look here: https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous